File: Vehicle.Projetas/Vehicle.Projetas.Web.Api/Controllers/VehicleController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;

namespace Vehicle.Projetas.Web.Api.Controllers
{
    public class VehicleController : ApiController
    {
        Bll.Vehicle vehicleBll = new Bll.Vehicle();

        public IEnumerable<Model.Vehicle> Get()
        {
            try
            {
                return vehicleBll.readAll();
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex.InnerException);
            }
        }

        public IEnumerable<Model.Vehicle> Get(int id)
        {
            try
            {
                return vehicleBll.read(v => v.Id == id);
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex.InnerException);
            }
        }

        public void Post([FromBody]Model.Vehicle vehicle)
        {
            try
            {             
                vehicleBll.create(vehicle);
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex.InnerException);
            }
        }

        public void Put(int id, [FromBody]Model.Vehicle vehicle)
        {
            try
            {                
                vehicleBll.update(vehicle);
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex.InnerException);
            }
        }

        public void Delete(int id)
        {
            try
            {
                vehicleBll.delete(v => v.Id == id);
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message, ex.InnerException);
            }
        }
    }
}

File: backend/src/main/java/br/com/projetas/jobs/controllers/VehicleController.java
package br.com.projetas.jobs.controllers;

import br.com.projetas.jobs.model.Vehicle;
import br.com.projetas.jobs.services.VehicleService;
import org.springframework.web.bind.annotation.*;

@RestController
@CrossOrigin(origins = "http://localhost:3000")
public class VehicleController {

    private VehicleService service;

    public VehicleController(VehicleService service) {
        this.service = service;
    }

    @RequestMapping(path = "/vehicles", method = RequestMethod.GET)
    public Iterable<Vehicle> list() {
        return service.list();
    }

    @RequestMapping(path = "/vehicle", method = RequestMethod.POST)
    public void create(@RequestBody Vehicle vehicle) {
        service.create(vehicle);
    }

    @RequestMapping(path = "/vehicle/{id}", method = RequestMethod.PUT)
    public void update(@RequestBody Vehicle vehicle, @PathVariable Long id) {
        service.update(vehicle, id);
    }

    @RequestMapping(path = "/vehicle/{id}", method = RequestMethod.DELETE)
    public void delete(@PathVariable Long id) {
        service.delete(id);
    }

    @RequestMapping(path = "/vehicle/{id}", method = RequestMethod.GET)
    public Vehicle findById(@PathVariable Long id) {
        return service.findById(id);
    }
}
